### PR TITLE
Tower refactor

### PR DESCRIPTION
**Pull request by 0o-de-lally**
Refactor of the tower proof submission, including plumbing from Libra repo. Mainly changes the looping logic for proofs and fetching credentials for proof transactions. Also scaffolds some views for common sync and tower error conditions.

Also includes a rudimentary sketch of CarpeError generation, and a simple state machine client side for errors.

### CODE REVIEW

File: src-tauri/src/carpe_error.rs [hunk 1]
```diff
@@ -24,30 +26,62 @@ impl From<anyhow::Error> for CarpeError {
     }
 }
 
-const E_MISC: u8 = 100;
+impl From<TxError> for CarpeError {
+    fn from(e: TxError) -> Self {
+      // check if the is a tower error
+      match tower::tower_errors::parse_error(e){
+        tower::tower_errors::TowerError::Unknown => CarpeError::tx("unknown tx error submitting tower"),
+        tower::tower_errors::TowerError::Other(v) => CarpeError::tx(&format!("unknown tx error submitting tower, message: {}", &v)),
```
Comment (liangping) [inline src-tauri/src/carpe_error.rs:34]: Wondering if unknown and Other could combined into one, It's confusing somehow.

File: src-tauri/src/commands/mining.rs [hunk 2]
```diff
@@ -108,89 +112,71 @@ fn get_proof_zero() -> Result<VDFProof, Error> {
...
+  let v = env::var("NODE_ENV")
+    .map_err(|_| CarpeError::misc("environment variable NODE_ENV is not set"))?;
   Ok(v)
 }
 
-#[tauri::command]
+#[tauri::command(async)]
 pub fn get_env() -> Result<String, CarpeError> {
-  let v = env::var("NODE_ENV").map_err(|_| { CarpeError::misc("environment variable NODE_ENV is not set") })?;
+  let v = env::var("NODE_ENV")
```
Comment (liangping) [inline src-tauri/src/commands/mining.rs:179]: I am prefer to return a `prod` as default value.  if so, we will reduce unnecessary error handling code.

**Review by soaresa — APPROVED** [review]
It may need some adjustments, but let's merge and keep the work flowing till a next version release.

File: src-tauri/src/carpe_error.rs (hunk 1 above)
Comment (0o-de-lally) [reply src-tauri/src/carpe_error.rs:34]: yes this would be a change in Libra repo though.

File: src-tauri/src/commands/mining.rs (hunk 2 above)
Comment (0o-de-lally) [reply src-tauri/src/commands/mining.rs:179]: OK done.

**Pull request merged**